src/combinatorics/simple_graph/matching.lean

```diff
@@ -56,6 +56,18 @@ contained in some edge of matching `M`
 def matching.support (M : G.matching) : set V :=
 {v : V | ∃ x ∈ M.edges, v ∈ x}
 
+/--
+A vertex `v` is saturated by a matching `M` if `v ∈ M.support`.
+-/
+def matching.saturated_vertex (M : G.matching) (v : V) : Prop :=
+v ∈ M.support
```

Comment:
Bear in mind that `M.saturated_vertex` can be used as a set as well if you change the definition to
```
def matching.saturated_vertex (M : G.matching) : set V :=
λ v, v ∈ M.support
```
But then at that point you might as well just say 
```
def matching.saturated_vertex (M : G.matching) : set V :=
M.support
```

Perhaps then it might be cleaner to change `support` to be `saturated` or something, and so the syntax would be `M.saturated v` or `v ∈ M.saturated`. What do you and others think?

Reply:
Ohh that could be interesting actually

Reply:
I've been told that `M.saturated v` is considered to be abusing the `set` interface (otherwise the neighbor set for simple graphs might have just been `G.adj v`).

I'm not sure `matching.saturated_vertex` is a particularly important concept compared to `matching.saturated_set`.  I'd suggest removing it and then renaming the latter to `matching.saturates`, so then `M.saturates S` is quicker to comprehend.

Reply:
So would 
```
/--
A set of vertices `S` is saturated by a matching `M` if `S ⊆ M.support`
-/
def matching.saturates (M : G.matching) (S : set V) : Prop :=
S ⊆ M.support
```

be an okay way of defining things? To the point @b-mehta made, I realized that the idea of changing the name of `support` to `saturates` isn't the same because the idea is that a subset of the vertex set might be saturated. But @kmill pointed out that it might be better to just not have the definition altogether because it would ruin the effectiveness of `simp` or something

Reply:
I just did an experiment with the code on that branch for the Hall marriage theorem, switching out all cases of `M.saturates S` for `S ⊆ M.support`, and I think it's actually clearer that way, and some of the lemmas can be removed entirely.

My vote right now is for not having either definition.

(If you don't like the name `support`, another name could be `matched` for the set of matched vertices.)

Reply:
Perhaps it's worth adding to the docstring of `support` to include how the definition relates to the concept of `saturation`?

Reply:
@kmill When you say either definition, do you mean the two saturated definitions?

Reply:
@agusakov Yeah, I mean having neither a definition for saturated vertices nor saturated sets.

Reply:
What's the conclusion here -- are we waiting for changes based on this discussion?

Reply:
Sorry about that - I think I'm just going to abandon the suggested changes